### ISSUE TITLE
Translated route not redirecting if current locale does not match.

### ISSUE THREAD

**Issue by istratos** [mcamara/laravel-localization#493]
Utilizing all 4 middleware for this procedure.

English route:
/contactus

Spanish route:
/es/contacto

If current locale is EN (English):
/contactus shows english page fine. (correct)
/en/contactus redirects to contactus (correct)
/contacto shows 404 (should redirect to /es/contacto)
/es/contacto shows spanish page fine. (correct)

If current locale is ES (Spanish):
/contactus redirects to es/contacto (should show english page not redirect to spanish)
/en/contactus shows english page fine (correct)
/contacto shows 404 (should redirect to /es/contacto)
/es/contacto shows spanish page fine. (correct)

So in short, the localize uri name won't ever work. And if I'm in a different locale, the unlocalized uri name redirects to the current localized uri.

Why is this?

Laravel 5.5
Laravel-localization 1.3.2

**Comment by dorinesinenco** [comment]
+1, Subscribe to this issue.
Same goes in my case, we have 2 projects where 
/test/{param}
/ru/тест/{param} works on default language
but when switching to the translated counterpart the route becomes
/тест/{param}
/ru/тест/{param}
laravel 5.5
Laravel-localization 1.3.2

**Comment by SandhaniDev** [comment]
im getting problem on auth redirection is not going localization name
im getting after register http://localhost:8000/profile
but i need like dis http://localhost:8000/en/profile

**Comment by FatihAydin52** [comment]
+1

**Comment by niels-numbers** [comment]
For untranslated routes, the idea behind calling a route without a locale is to let the app detect the correct locale and handle the redirect. In your case, you already know that `/contacto` is the Spanish version, so it would make more sense to link directly to `/es/contacto`.

We don't need the full complexity required for normal locale-less routes (checking browser locale, session, cookie, etc.). In your case, you could just loop through your `lang/**/routes.php` files and add simple redirects from `/contacto` to `/es/contacto`, and so on. That said, I'd argue it's probably better to never link to a translated route without its locale in the first place.
